<a href="https://colab.research.google.com/github/saja13-zh/sql_IBM/blob/main/sql_final_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet.

Use the links below to read the data files using the Pandas library.

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [ ]:
import pandas as pd
import sqlite3
con=sqlite3.connect("FinalDB.db")

Load the SQL magic module


In [ ]:
%load_ext sql

Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


In [ ]:
CENSUS_DATA=pd.read_csv("/content/ChicagoCensusData.csv")
CHICAGO_PUBLIC_SCHOOLS=pd.read_csv("/content/ChicagoPublicSchools.csv")
CHICAGO_CRIME_DATA=pd.read_csv("/content/ChicagoCrimeData.csv")

CENSUS_DATA.to_sql("CENSUS_DATA",con,index=False, if_exists='replace')
CHICAGO_PUBLIC_SCHOOLS.to_sql("CHICAGO_PUBLIC_SCHOOLS",con,index=False, if_exists='replace')
CHICAGO_CRIME_DATA.to_sql("CHICAGO_CRIME_DATA",con,index=False, if_exists='replace')

con.commit()


In [ ]:
CHICAGO_CRIME_DATA.head(20)

ID CASE_NUMBER        DATE                     BLOCK IUCR  \
0    3512276    HK587712  2004-08-28        047XX S KEDZIE AVE  890   
1    3406613    HK456306  2004-06-26  009XX N CENTRAL PARK AVE  820   
2    8002131    HT233595  2011-04-04        043XX S WABASH AVE  820   
3    7903289    HT133522  2010-12-30      083XX S KINGSTON AVE  840   
4   10402076    HZ138551  2016-02-02           033XX W 66TH ST  820   
5    7732712    HS540106  2010-09-29       006XX W CHICAGO AVE  810   
6   10769475    HZ534771  2016-11-30        050XX N KEDZIE AVE  810   
7    4494340    HL793243  2005-12-16       005XX E PERSHING RD  860   
8    3778925    HL149610  2005-01-28     100XX S WASHTENAW AVE  810   
9    3324217    HK361551  2004-05-13       033XX W BELMONT AVE  820   
10   9902691    HX552797  2014-12-23        076XX S CICERO AVE  860   
11   7947964    HT180228  2011-02-26        012XX S HALSTED ST  820   
12   8568365    HV243612  2012-04-15       108XX S EMERALD AVE  820   
13   5343947    HN200365  2007-02-26          011XX S CANAL ST  842   
14   3121541    HK108680  2003-01-05       038XX W FLOURNOY ST  890   
15  10209850    HY396729  2015-08-24        026XX N ELSTON AVE  820   
16   6866556    HR271603  2009-04-16  089XX S STONY ISLAND AVE  860   
17   9681782    HX332132  2014-07-04          079XX S STATE ST  820   
18   9166453    HW303994  2013-06-04        055XX W GLADYS AVE  810   
19   7762423    HS569246  2010-10-17     058XX W ARTHINGTON ST  880   

   PRIMARY_TYPE                    DESCRIPTION  \
0         THEFT                  FROM BUILDING   
1         THEFT                 $500 AND UNDER   
2         THEFT                 $500 AND UNDER   
3         THEFT  FINANCIAL ID THEFT: OVER $300   
4         THEFT                 $500 AND UNDER   
5         THEFT                      OVER $500   
6         THEFT                      OVER $500   
7         THEFT                   RETAIL THEFT   
8         THEFT                      OVER $500   
9         THEFT                 $500 AND UNDER   
10        THEFT                   RETAIL THEFT   
11        THEFT                 $500 AND UNDER   
12        THEFT                 $500 AND UNDER   
13        THEFT        AGG: FINANCIAL ID THEFT   
14        THEFT                  FROM BUILDING   
15        THEFT                 $500 AND UNDER   
16        THEFT                   RETAIL THEFT   
17        THEFT                 $500 AND UNDER   
18        THEFT                      OVER $500   
19        THEFT                PURSE-SNATCHING   

              LOCATION_DESCRIPTION  ARREST  DOMESTIC  ...  DISTRICT  WARD  \
0               SMALL RETAIL STORE   False     False  ...         9  14.0   
1                            OTHER   False     False  ...        11  27.0   
2     NURSING HOME/RETIREMENT HOME   False     False  ...         2   3.0   
3                        RESIDENCE   False     False  ...         4   7.0   
4                            ALLEY   False     False  ...         8  15.0   
5   PARKING LOT/GARAGE(NON.RESID.)   False     False  ...        12  27.0   
6                           STREET   False     False  ...        17  33.0   
7               GROCERY FOOD STORE    True     False  ...         2   3.0   
8                           STREET   False     False  ...        22  19.0   
9               SMALL RETAIL STORE   False     False  ...        17  35.0   
10                DEPARTMENT STORE    True     False  ...         8  13.0   
11                   BAR OR TAVERN   False     False  ...        12  25.0   
12   RESIDENTIAL YARD (FRONT/BACK)   False     False  ...        22  34.0   
13                           OTHER   False     False  ...         1   2.0   
14         RESIDENCE PORCH/HALLWAY   False     False  ...        11  24.0   
15  PARKING LOT/GARAGE(NON.RESID.)   False     False  ...        14   1.0   
16                           OTHER    True     False  ...         4   8.0   
17                    CTA PLATFORM   False     False  ...         6   6.0   
18  

In [ ]:
CENSUS_DATA.head(20)

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
0                     1.0         Rogers Park                         7.7   
1                     2.0          West Ridge                         7.8   
2                     3.0              Uptown                         3.8   
3                     4.0      Lincoln Square                         3.4   
4                     5.0        North Center                         0.3   
5                     6.0           Lake View                         1.1   
6                     7.0        Lincoln Park                         0.8   
7                     8.0     Near North Side                         1.9   
8                     9.0         Edison Park                         1.1   
9                    10.0        Norwood Park                         2.0   
10                   11.0      Jefferson Park                         2.7   
11                   12.0         Forest Glen                         1.1   
12                   13.0          North Park                         3.9   
13                   14.0         Albany Park                        11.3   
14                   15.0        Portage Park                         4.1   
15                   16.0         Irving Park                         6.3   
16                   17.0             Dunning                         5.2   
17                   18.0          Montclaire                         8.1   
18                   19.0      Belmont Cragin                        10.8   
19                   20.0             Hermosa                         6.9   

    PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
0                               23.6                          8.7   
1                               17.2                          8.8   
2                               24.0                          8.9   
3                               10.9                          8.2   
4                                7.5                          5.2   
5                               11.4                          4.7   
6                               12.3                          5.1   
7                               12.9                          7.0   
8                                3.3                          6.5   
9                                5.4                          9.0   
10                               8.6                         12.4   
11                               7.5                          6.8   
12                              13.2                          9.9   
13                              19.2                         10.0   
14                              11.6                         12.6   
15                              13.1                         10.0   
16                              10.6                         10.0   
17                              15.3                         13.8   
18                              18.7                         14.6   
19                              20.5                         13.1   

    PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
0                                           18.2   
1                                           20.8   
2                                           11.8   
3                                           13.4   
4                                            4.5   
5                                            2.6   
6                                            3.6   
7                                            2.5   
8                                            7.4   
9                                           11.5   
10                                          13.4   
11                                           4.9   
12                                          14.4   
13                                          32.9   
14                                          19.3   
15                                          22.4   
16                                          16.2   


In [ ]:
CHICAGO_PUBLIC_SCHOOLS.head()

School_ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School         Street_Address     City State  \
0                                 ES        615 W Kemper Pl  Chicago    IL   
1                                 ES  7511 S South Shore Dr  Chicago    IL   
2                                 ES     8010 S Kostner Ave  Chicago    IL   
3                                 ES     4619 S Wolcott Ave  Chicago    IL   
4                                 HS        3630 S Wells St  Chicago    IL   

   ZIP_Code    Phone_Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                                Link  \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network_Manager  ... Freshman_on_Track_Rate__  \
0        Fullerton Elementary Network  ...                      NDA   
1           Skyway Elementary Network  ...                      NDA   
2           Midway Elementary Network  ...                      NDA   
3         Pershing Elementary Network  ...                      NDA   
4  Southwest Side High School Network  ...                     91.8   

  X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0  1171699.458  1915829.428  41.924497 -87.644522                     7   
1  1196129.985  1856209.466  41.760324 -87.556736                    43   
2  1148427.165  1851012.215  41.747111 -87.731702                    70   
3  1164504.290  1873959.199  41.809757 -87.672145                    61   
4  1175177.622  1880745.126  41.828146 -87.632794                    34   

  COMMUNITY_AREA_NAME  Ward Police_District                     Location  
0        LINCOLN PARK    43              18  (41.92449696, -87.64452163)  
1         SOUTH SHORE     7               4  (41.76032435, -87.55673627)  
2             ASHBURN    13               8  (41.74711093, -87.73170248)  
3            NEW CITY    20               9    (41.8097569, -87.6721446)  
4       ARMOUR SQUARE    11               9  (41.82814609, -87.63279369)  

[5 rows x 78 columns]

Establish a connection between SQL magic module and the database `FinalDB.db`


In [ ]:
cur = con.cursor()

You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [ ]:
q1=cur.execute("SELECT count(*) FROM CHICAGO_CRIME_DATA")
q1.fetchall()

[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [ ]:
q2=cur.execute("SELECT COMMUNITY_AREA_NUMBER ,COMMUNITY_AREA_NAME FROM CENSUS_DATA WHERE PER_CAPITA_INCOME<11000 ")
q2.fetchall()

[(26.0, 'West Garfield Park'),
 (30.0, 'South Lawndale'),
 (37.0, 'Fuller Park'),
 (54.0, 'Riverdale')]

### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis)


In [ ]:
q3=cur.execute("SELECT CASE_NUMBER FROM CHICAGO_CRIME_DATA WHERE DESCRIPTION like 'mino%'  ")
q3.fetchall()

[]

### Problem 4

##### List all kidnapping crimes involving a child?


In [ ]:
q4=cur.execute("SELECT * FROM CHICAGO_CRIME_DATA WHERE PRIMARY_TYPE like 'KIDNAPPING%'")
q4.fetchall()

[(5276766,
  'HN144152',
  '2007-01-26',
  '050XX W VAN BUREN ST',
  '1792',
  'KIDNAPPING',
  'CHILD ABDUCTION/STRANGER',
  'STREET',
  0,
  0,
  1533,
  15,
  29.0,
  25.0,
  '20',
  1143050.0,
  1897546.0,
  2007,
  41.87490841,
  -87.75024931,
  '(41.874908413, -87.750249307)')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [ ]:
q5=cur.execute("SELECT DISTINCT PRIMARY_TYPE FROM CHICAGO_CRIME_DATA WHERE LOCATION_DESCRIPTION like 'scho%' ")
q5.fetchall()

[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [ ]:
q6 = cur.execute("SELECT \"Elementary, Middle, or High School\", AVG(SAFETY_SCORE) FROM CHICAGO_PUBLIC_SCHOOLS GROUP BY \"Elementary, Middle, or High School\" ")
q6.fetchall()

[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [ ]:
q7=cur.execute("SELECT PERCENT_HOUSEHOLDS_BELOW_POVERTY	,COMMUNITY_AREA_NAME FROM CENSUS_DATA order by PERCENT_HOUSEHOLDS_BELOW_POVERTY desc limit 5  ")
q7.fetchall()

[(56.5, 'Riverdale'),
 (51.2, 'Fuller Park'),
 (46.6, 'Englewood'),
 (43.1, 'North Lawndale'),
 (42.4, 'East Garfield Park')]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [ ]:
q8=cur.execute("SELECT COMMUNITY_AREA_NUMBER, count(COMMUNITY_AREA_NUMBER) AS IncidentCount FROM CHICAGO_CRIME_DATA  GROUP BY COMMUNITY_AREA_NUMBER ORDER BY IncidentCount DESC LIMIT 1")
q8.fetchall()

[(25.0, 43)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [ ]:
q9=cur.execute("SELECT COMMUNITY_AREA_NAME FROM CENSUS_DATA WHERE HARDSHIP_INDEX=(SELECT max(HARDSHIP_INDEX) FROM CENSUS_DATA) ")
q9.fetchall()

[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [ ]:
q10=cur.execute("SELECT COMMUNITY_AREA_NAME FROM CENSUS_DATA where COMMUNITY_AREA_NUMBER= (SELECT COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA group by COMMUNITY_AREA_NUMBER order by COUNT(*) DESC limit 1  ) ")
q10.fetchall()

[('Austin',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
